File: arven-towns-api/src/main/java/arvenwood/towns/api/town/TownService.java
package arvenwood.towns.api.town;

import org.spongepowered.api.Sponge;

import java.util.Collection;
import java.util.Optional;
import java.util.UUID;

public interface TownService {

    static TownService get() {
        return Sponge.getServiceManager().provideUnchecked(TownService.class);
    }

    Collection<Town> getTowns();

    Optional<Town> getTown(UUID uniqueId);

    Optional<Town> getTown(String name);

    boolean contains(Town town);

    boolean register(Town town);

    boolean unregister(Town town);
}

File: src/main/kotlin/arvenwood/towns/api/town/TownService.kt
package arvenwood.towns.api.town

import org.spongepowered.api.Sponge
import java.util.*

interface TownService {

    companion object {
        @JvmStatic
        fun get(): TownService =
            Sponge.getServiceManager().provideUnchecked(TownService::class.java)
    }

    val towns: Collection<Town>

    fun getTown(uniqueId: UUID): Town?

    fun getTown(name: String): Town?

    fun register(town: Town): Boolean

    fun unregister(town: Town): Boolean
}